### PR TITLE
Add support for inlined identity comparisons

### PR DESCRIPTION
**Pull request by guillep**
Make `bytecodePrim[Not]IdenticalSistaV1` not generate a message send

### CODE REVIEW

File: smalltalksrc/VMMaker/CoInterpreter.class.st [hunk 1]
```diff
@@ -2751,28 +2751,6 @@ CoInterpreter >> executeNewMethod: eagerlyCompile [
 		self activateNewMethod ]
 ]
 
-{ #category : #'message sending' }
-CoInterpreter >> executeNewMethod: eagerlyCompile [
```
Comment (PalumboN) [inline smalltalksrc/VMMaker/CoInterpreter.class.st:2755]: This method was installed twice in the same class?? 🤯

File: smalltalksrc/VMMaker/StackInterpreter.class.st [hunk 2]
```diff
@@ -2624,55 +2624,61 @@ StackInterpreter >> binarySmiCompInlinePrimitive: primIndex [
...
+	resultBoolean ifNotNil: [ self push: resultBoolean ]
 ]
 
 { #category : #utilities }
 StackInterpreter >> booleanCheatTrueSistaV1 [
 	"cheat the interpreter out of the pleasure of handling the next bytecode IFF it is a jump-on-boolean. Which it is, often enough when the current bytecode is something like bytecodePrimEqual"
 
-	<sharedCodeInCase: #bytecodePrimLessThanSistaV1>
-	self pop: 2.
+	"<sharedCodeInCase: #bytecodePrimLessThanSistaV1>"
```
Comment (PalumboN) [inline smalltalksrc/VMMaker/StackInterpreter.class.st:2662]: This is not necessary? Or did you comment the line for a test?

File: smalltalksrc/VMMaker/StackInterpreter.class.st [hunk 3]
```diff
@@ -2984,19 +2990,14 @@ StackInterpreter >> bytecodePrimGreaterThanSistaV1 [
...
-		^ self booleanCheatSistaV1: rcvr = arg ].
-
-  self normalSendSpecialSelector: 22 argumentCount: 1
+	rcvr := self stackValue: 1.
+	arg := self stackValue: 0.
+	(objectMemory isOopForwarded: rcvr) ifTrue: [
+		rcvr := self handleSpecialSelectorSendFaultFor: rcvr ].
+	(objectMemory isOopForwarded: arg) ifTrue: [
+		arg := self handleSpecialSelectorSendFaultFor: arg ].
+	^ self booleanCheatSistaV1: rcvr = arg
```
Comment (PalumboN) [inline smalltalksrc/VMMaker/StackInterpreter.class.st:3000]: So these bytecodes never send the message?

**Review by PalumboN — APPROVED** [review]
🅾️ = ⭕

File: smalltalksrc/VMMaker/CoInterpreter.class.st (hunk 1 above)
Comment (guillep) [reply smalltalksrc/VMMaker/CoInterpreter.class.st:2755]: NO IDEA. Maybe a merge problem?

Comment (guillep) [reply smalltalksrc/VMMaker/CoInterpreter.class.st:2755]: My super good PR fixes it ;)

File: smalltalksrc/VMMaker/StackInterpreter.class.st (hunk 2 above)
Comment (guillep) [reply smalltalksrc/VMMaker/StackInterpreter.class.st:2662]: The pragma is necessary for slang translation.
But with my changes such a translation is not valid anymore.
We should probably discuss this better.

File: smalltalksrc/VMMaker/StackInterpreter.class.st (hunk 3 above)
Comment (guillep) [reply smalltalksrc/VMMaker/StackInterpreter.class.st:3000]: It never did in stock VMs. Only in druid.

**Pull request merged**